packages/cli/src/oclif/commands/validate.js

```diff
@@ -0,0 +1,108 @@
+const colors = require('colors/safe');
+const { flags } = require('@oclif/command');
+
+const BaseCommand = require('../ZapierBaseCommand');
+const utils = require('../../utils');
+const { buildFlags } = require('../buildFlags');
+const { flattenCheckResult, formatStyles } = require('../../utils/display');
+
+class ValidateCommand extends BaseCommand {
+  async perform() {
+    this.log('Validating project locally.');
+
+    const errors = await utils.localAppCommand({ command: 'validate' });
+
+    const newErrors = errors.map(error => {
+      error = Object.assign({}, error);
+      error.property = error.property.replace(/^instance/, 'App');
+      error.docLinks = (error.docLinks || []).join('\n');
+      return error;
+    });
+    this.logTable({
+      rows: newErrors,
+      headers: [
+        ['Property', 'property'],
+        ['Message', 'message'],
+        ['Links', 'docLinks']
+      ],
+      emptyMessage: colors.grey(
+        'No structural errors found during validation routine.'
+      )
+    });
+
+    if (newErrors.length) {
+      this.log(
+        'Your app is structurally invalid. Address concerns and run this command again.'
+      );
+      process.exitCode = 1;
+    } else {
+      this.log('This project is structurally sound!');
+    }
+
+    let checkResult = [];
+    if (0 && (this.flags['no-server'] || process.exitCode === 1)) {
+      if (process.exitCode === 1) {
+        this.log(
+          colors.grey('\nSkipping app checks because app did not validate.')
+        );
+      }
+      return;
+    } else {
+      this.log('\nRunning app checks.');
+
+      const rawDefinition = await utils.localAppCommand({
+        command: 'definition'
+      });
+
+      checkResult = await utils.validateApp(rawDefinition);
+    }
+
+    const checkIssues = flattenCheckResult(checkResult);
+
+    this.logTable({
+      rows: checkIssues,
+      headers: [
+        ['Category', 'category'],
+        ['Method', 'method'],
+        ['Description', 'description'],
+        ['Link', 'link']
+      ],
+      emptyMessage: colors.grey('App checks passed, no issues found.')
+    });
+
+    if (checkResult.errors && checkResult.errors.length) {
+      process.exitCode = 1;
+      this.log(
+        'Errors will block you from pushing; warnings only block you from going public or promoting a version.\n'
+      );
+    } else if (checkResult.warnings.length) {
+      this.log(
+        'Your app looks great! Warnings only block you from going public or promoting a version.\n'
+      );
+    } else {
+      this.log('Your app looks great!\n');
+    }
+  }
+}
+
+ValidateCommand.flags = buildFlags({
+  commandFlags: {
+    format: flags.string({
```

Comment:
Yeah, it was repeated because I wanted to give it a different default ("row" rather than "table"). But on the second thought, I think it's fine to use the default default format of "table" because "table" format becomes "row" automatically when the screen width doesn't fit, which is very likely to happen. Fixed in 1632a8c.

Reply:
ok! we can also make the default configurable if that's a thing we find useful elsewhere.